.github/actions/upload/action.yml

```diff
@@ -29,3 +33,11 @@ runs:
         path: |
           **/surefire-reports/TEST-*.xml
         retention-days: ${{ inputs.test-report-retention-days }}
+    - name: Upload heap dump
+      uses: actions/upload-artifact@v3
+      if: failure() && inputs.upload-heap-dump
+      with:
+        name: heap dump ${{ github.job }}
+        path: |
+          **/*.hprof
```

Comment:
Will this get compressed?
heap dumps are extremely well compressible, right?

Reply:
All artifacts are zipped.